README.rst

```diff
@@ -141,129 +245,1147 @@ You can immediately use Ivy to train a neural network, using your favorite frame
-but the backend can easily be changed to your favorite frameworks, such as TensorFlow, or JAX.
 
-**Framework Agnostic Functions**
+Last but no least, we are also working on specific extension totally written in Ivy and therefore usable within any framework, 
+covering topics like `Mechanics`_, `Computer Vision`_, `Robotics`_, a `Reinforcement Learning Gym`_, `Memory`_ and implementation of various `Models`_ or `Builder tools`_ with trainers, data loaders and more.
+
+TODO: Maybe add small logos in a row?
+
+As always, you can find more information about `Ivy as a framework in the docs! <https://lets-unify.ai/ivy/design/ivy_as_a_framework.html#ivy-as-a-framework>`_
+
+When should I use Ivy as a framework?
+######################################
+
+As Ivy supports multiple backends, writing code in Ivy breaks you free from framework limitations. If you want to publish highly flexible code for everyone to use, independently of the framework they are using, or you plan to develop ML-related tools and want them to be interoperable with not only the already existing frameworks, but also with future frameworks, then Ivy is for you!
+
+Setting up Ivy
+--------------
+
+ToDo: test each one of these procedures in various platforms, add links
+
+There are various ways to use Ivy, depending on your preferred environment:
+
+Installing using pip
+####################
+
+The easiest way to set up Ivy is to install it using pip with the following command:
+
+.. code-block:: bash
+
+    pip install ivy-core
+
+or alternatively:
+
+.. code-block:: bash
+
+    python3 -m pip install ivy-core
+
+
+Docker
+######
+
+If you prefer to use containers, we also have pre-built Docker images with all the supported frameworks and some relevant packages already installed, which you can pull from:
+
+.. code-block:: bash
+
+    docker pull unifyai/ivy:latest
+
+If you are working on a GPU device, you can pull from:
+
+.. code-block:: bash
+
+    docker pull unifyai/ivy:latest-gpu
+
+Installing from source
+######################
+
+Obviously, you can also install Ivy from source if you want to take advantage of the latest changes, but we can't ensure that everything will work as expected :sweat_smile:
+
+.. code-block:: bash
+
+    git clone https://github.com/unifyai/ivy.git
+    cd ivy 
+    pip install --user -e .
+
+or alternatively, for the last step:
+
+.. code-block:: bash
+
+    python3 -m pip install --user -e .
+
+If you want to set up testing and various frameworks it's probably best to check out the `Contributing - Setting Up <https://lets-unify.ai/ivy/contributing/setting_up.html#setting-up>`_ page, where OS-specific and IDE-specific instructions and video tutorials to do so are available!
 
-In the example below we show how Ivy's concatenation function is compatible with tensors from different frameworks.
-This is the same for ALL Ivy functions. They can accept tensors from any framework and return the correct result.
+
+Access to the Transpiler API
```

Comment:
Let's remove all of this for now. We will be releasing a fully ungated binary publicly now, so there is no need to have this on the readme :)